elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemSchema.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.extension.elytron;
+
+import org.jboss.as.controller.AttributeMarshallers;
+import org.jboss.as.controller.AttributeParsers;
+import org.jboss.as.controller.Feature;
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.PersistentSubsystemSchema;
+import org.jboss.as.controller.SubsystemSchema;
+import org.jboss.as.controller.xml.VersionedNamespace;
+import org.jboss.as.version.Stability;
+import org.jboss.staxmapper.IntVersion;
+
+import java.util.EnumSet;
+import java.util.Map;
+
+import static org.wildfly.extension.elytron.ElytronDescriptionConstants.SECURITY_PROPERTY;
+
+/**
+ * Enumeration of elytron subsystem schema versions.
+ */
+public enum ElytronSubsystemSchema implements PersistentSubsystemSchema<ElytronSubsystemSchema> {
+    VERSION_1_0(1),
+    VERSION_1_1(1, 1),
+    VERSION_1_2(1, 2),
+    VERSION_2_0(2),
+    VERSION_3_0(3),
+    VERSION_4_0(4),
+    VERSION_5_0(5),
+    VERSION_6_0(6),
+    VERSION_7_0(7),
+    VERSION_8_0(8),
+    VERSION_9_0(9),
+    VERSION_10_0(10),
+    VERSION_11_0(11),
+    VERSION_12_0(12),
+    VERSION_13_0(13),
+    VERSION_14_0(14),
+    VERSION_15_0(15),
+    VERSION_15_1(15, 1),
+    VERSION_16_0(16),
+    VERSION_17_0(17),
+    VERSION_18_0(18),
+    VERSION_18_0_COMMUNITY(18, Stability.COMMUNITY),
+    ;
+    static final Map<Stability, ElytronSubsystemSchema> CURRENT = Feature.map(EnumSet.of(VERSION_18_0, VERSION_18_0_COMMUNITY));
+
+    private final VersionedNamespace<IntVersion, ElytronSubsystemSchema> namespace;
+
+    static final ElytronSubsystemParserCommunity18_0 latestParser = new ElytronSubsystemParserCommunity18_0();
+
+    ElytronSubsystemSchema(int major) {
+        this.namespace = SubsystemSchema.createSubsystemURN(ElytronExtension.SUBSYSTEM_NAME, new IntVersion(major));
+    }
+
+    ElytronSubsystemSchema(int major, int minor) {
+        this.namespace = SubsystemSchema.createSubsystemURN(ElytronExtension.SUBSYSTEM_NAME, new IntVersion(major, minor));
+    }
+
+    ElytronSubsystemSchema(int major, Stability stability) {
+        this.namespace = SubsystemSchema.createSubsystemURN(ElytronExtension.SUBSYSTEM_NAME, stability, new IntVersion(major));
+    }
+
+    @Override
+    public VersionedNamespace<IntVersion, ElytronSubsystemSchema> getNamespace() {
+        return this.namespace;
+    }
+
+    @Override
+    public PersistentResourceXMLDescription getXMLDescription() {
```

Comment:
This doesn't look right.  Why does every version use the same PersistentResourceXMLDescription?

Reply:
@pferraro Thanks! I fixed this and used a new ElytronPersistentResourceXMLDescriptionFactory class here